integration/spark2/src/main/scala/org/apache/spark/sql/CarbonEnv.scala

```diff
@@ -125,6 +127,19 @@ class CarbonEnv {
     LOGGER.info("Initialize CarbonEnv completed...")
   }
 
+
+  def createIndexServerTmp(sparkSession: SparkSession): Unit = {
```

Comment:
We already have a method to create temp folder in CarbonUtil. Please use the same method and add the code to change the permission to 777

Reply:
done